Windows build fails with "invalid numeric argument '/Wno-unused-parameter'"

cl : Command line error D8021 : invalid numeric argument '/Wno-unused-parameter'
cl : Command line error D8021 : invalid numeric argument '/Wno-unused-parameter'
make[2]: *** [lib/CoreLibraries.gmk:49: /cygdrive/d/a/icedtea8/icedtea8/jdk/build/windows-x64/jdk/objs/libfdlibm/e_acos.obj] Error 1

--- a/jdk/make/CompileNativeLibraries.gmk
+++ b/jdk/make/CompileNativeLibraries.gmk
@@ -42,9 +42,11 @@ $(eval $(call FillCacheFind, $(JDK_TOPDIR)/src))
 include Tools.gmk
 
 # Handle warnings appropriately
-WARNING_CFLAGS = -Wno-unused-parameter
-ifeq ($(USE_CLANG), true)
-  WARNING_CFLAGS += -Qunused-arguments
+ifneq ($(filter $(TOOLCHAIN_TYPE), gcc clang), )
+  WARNING_CFLAGS = -Wno-unused-parameter
+  ifeq ($(USE_CLANG), true)
+    WARNING_CFLAGS += -Qunused-arguments
+  endif
 endif
 
 # Include the javah generated headers.

